### DISCUSSION TITLE
Macro Organization with Folders

### DISCUSSION THREAD

**Discussion by ttroutwine** [TabularEditor/TabularEditor3#1338] [Ideas]
### Is your feature request related to a problem? Please describe

Macros are organized sequentially by when added to TE3.  Its hard to group similar Macros together that perform similar operations.

### Describe the solution you'd like

It would beneficial to have the ability to group Macros in a Folder (better yet nested Sub Folder) within Tabular Editor 3.  It would be handy to have a Folder named Formatting to place my Macros for Power Querry Formatting and Format All Things.  A folder for Macros that perform documentation actions. 

### Goals

1.Folders in Macros
2.

### Background

As a work around I prefix the name with grouping categories and then sort macros by name.

**Comment by mlonsk** [comment]
Hi @ttroutwine 

Macros can be assigned to folders in the TOM Explorer by using the backslashes as show bellow. 
Is this what you were thinking of or do you have something else in mind?

![image](https://github.com/TabularEditor/TabularEditor3/assets/11939114/09ba8b13-84d0-4ea0-941c-4fad6ea386c0)

**Comment by ttroutwine** [comment]
Thank you, I tried this but didn’t look in the right spot to see if it was applied.

I was thinking along the lines of folders within the Macros window itself.  I didn’t realize backslashes would add the folder structure within the TOM Explorer.

I still think it would be beneficial to have the folders within the Macros window, but this definitely helps me.

Cody Troutwine | Business Intelligence Principal Engineer
BI-Engineering
***@***.******@***.***>

From: Morten Lønskov ***@***.***>
Sent: Friday, June 28, 2024 5:32 AM
To: TabularEditor/TabularEditor3 ***@***.***>
Cc: Cody Troutwine ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [TabularEditor/TabularEditor3] Macro Organization with Folders (Discussion #1338)

Caution: External email, beware of potential harmful links or attachments. If you don't recognize sender, report it to Carhartt Cyber Security using the 'Report Phish' button.

Hi @ttroutwine<https://urldefense.com/v3/__https:/github.com/ttroutwine__;!!JK3GimbO_w!iuP_FvYJArB-fJkGMOkzXBQkPo4rUdSNkgWhcIfx4JZAS2yfbdz9mGbXJGpHmSrIpwAudGwFZx509eusEcnJgDkEcCU$>

Macros can be assigned to folders in the TOM Explorer by using the backslashes as show bellow.
Is this what you were thinking of or do you have something else in mind?

image.png (view on web)<https://urldefense.com/v3/__https:/github.com/TabularEditor/TabularEditor3/assets/11939114/09ba8b13-84d0-4ea0-941c-4fad6ea386c0__;!!JK3GimbO_w!iuP_FvYJArB-fJkGMOkzXBQkPo4rUdSNkgWhcIfx4JZAS2yfbdz9mGbXJGpHmSrIpwAudGwFZx509eusEcnJnN01IlI$>

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/TabularEditor/TabularEditor3/discussions/1338*discussioncomment-9903017__;Iw!!JK3GimbO_w!iuP_FvYJArB-fJkGMOkzXBQkPo4rUdSNkgWhcIfx4JZAS2yfbdz9mGbXJGpHmSrIpwAudGwFZx509eusEcnJHMfRDJM$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/BJOHSMN2GPG7ALJE2B2SXBLZJUURXAVCNFSM6AAAAABKAP5WQCVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TSMBTGAYTO__;!!JK3GimbO_w!iuP_FvYJArB-fJkGMOkzXBQkPo4rUdSNkgWhcIfx4JZAS2yfbdz9mGbXJGpHmSrIpwAudGwFZx509eusEcnJWLjtta4$>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>

Confidential: This electronic message and all contents contain information from Carhartt, Inc. and its affiliates which may be confidential or otherwise protected from disclosure. The information is intended to be for the addressee only. If you are not the addressee, any disclosure, copy, distribution or use of the contents of this message is prohibited. If you have received this electronic message in error, please notify us immediately at (313) 271-8460 and destroy the original message and all copies.
